### PR TITLE
 drivers: wifi: Modify channel input for scan

### PR DESCRIPTION
**Pull request by rado17**

### CODE REVIEW

File: samples/wifi/shell/prj.conf [hunk 1]
```diff
@@ -67,6 +67,7 @@ CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN=y
 CONFIG_SHELL_CMDS_RESIZE=n
 #CONFIG_DEBUG=y
 CONFIG_WPA_SUPP_LOG_LEVEL_INF=y
+CONFIG_NET_IF_LOG_LEVEL_ERR=y
```
[the remark this answers is not in the document]
Comment (rado17) [reply samples/wifi/shell/prj.conf]: Removed

File: drivers/wifi/nrf700x/src/wifi_mgmt_scan.c [hunk 2]
```diff
@@ -117,6 +115,8 @@ int nrf_wifi_disp_scan_zep(const struct device *dev, struct wifi_scan_params *pa
 					     sizeof(scan_info->scan_params.center_frequency[0])));
 		goto out;
 	}
+	memset(scan_info, 0, sizeof(*scan_info) + (num_scan_channels *
```
[the remark this answers is not in the document]
Comment (rado17) [reply drivers/wifi/nrf700x/src/wifi_mgmt_scan.c:119]: Done

File: drivers/wifi/nrf700x/src/wifi_mgmt_scan.c [hunk 3]
```diff
@@ -146,32 +146,25 @@ int nrf_wifi_disp_scan_zep(const struct device *dev, struct wifi_scan_params *pa
...
-								    band,
-								    params->chan[i][j]);
-
-				if (scan_info->scan_params.center_frequency[k-1] == -1) {
-					LOG_ERR("%s: Invalid channel %d",
-						__func__,
-						params->chan[i][j]);
-					goto out;
-				}
+		for (j = 0; j < CONFIG_WIFI_MGMT_SCAN_CHAN_MAX_MANUAL; j++) {
```
[the remark this answers is not in the document]
Comment (rado17) [reply drivers/wifi/nrf700x/src/wifi_mgmt_scan.c]: Done

**Review by krish2718 — APPROVED** [review]

**Review by kapbh — APPROVED** [review]

**Comment by rado17** [comment]
Closing this as these changes have been committed to upmerge branch (https://github.com/de-nordic/fw-nrfconnect-nrf/pull/51)

**Pull request closed without merging**